staging/cse/windows/provisioningscripts/windowsnodereset.ps1

```diff
@@ -15,7 +15,12 @@ $global:NetworkMode = "L2Bridge"
 $global:NetworkPlugin = $Global:ClusterConfiguration.Cni.Name
 $global:ContainerRuntime = $Global:ClusterConfiguration.Cri.Name
 $UseContainerD = ($global:ContainerRuntime -eq "containerd")
-$IsDualStackEnabled = $Global:ClusterConfiguration.Kubernetes.Kubeproxy.FeatureGates -contains "IPv6DualStack=true"
+# if dual-stack is enabled, the clusterCidr will have an IPv6 CIDR in the comma separated list
+# we can split the entire string by ":" to get a count of how many ":" there are. If there are
+# at least 3 groups (which means there are at least 2 ":") then we know there is an IPv6 CIDR
+# in the list. We cannot just rely on `ClusterCidr -like "*::*" because there are IPv6 CIDRs that
+# don't have "::", e.g. fe80:0:0:0:0:0:0:0/64
+$IsDualStackEnabled = ($Global:ClusterConfiguration.Kubernetes.Network.ClusterCidr -split ":").Count -ge 3
 
```

Comment:
Q: Do we still need `featureGates["IPv6DualStack"] = true` with this change? I think that IPv6DualStack is not used by any feature except dual-stack

Reply:
probably not, but I'll keep it for now just in case